evm/src/executor/fork/cache.rs

```diff
@@ -204,14 +204,14 @@ impl MemDb {
                 accounts.insert(add, acc.info);
 
                 let acc_storage = storage.entry(add).or_default();
-                if acc.filth.abandon_old_storage() {
+                if acc.is_destroyed || acc.is_touched {
```

Comment:
@rakita @ngotchac wdyt here?

Reply:
Actually I think the right test might be `acc.is_destroyed || acc.storage_cleared` which are set in `create_account` or `selfdestruct` respectively which matches the previous behavior (cf. https://github.com/bluealloy/revm/pull/175/files)

Reply:
Only storage_cleared should be here is_destroyed is already checked above.